python/lsst/source/injection/inject_engine.py

```diff
@@ -542,6 +558,42 @@ def inject_galsim_objects_into_exposure(
                         err,
                     )
                 continue
+
+            # Set a variance level in each pixel
+            # corresponding to the drawn light profile.
+            variance_template = image_template.copy()
+            variance_template *= variance_scale
+
+            if add_noise:
+                # For generating noise,
+                # variance must be meaningful.
+                if np.any(variance_template.array < 0):
+                    if logger:
+                        logger.debug("Setting negative-variance pixels to 0 for noise generation.")
+                    variance_template.array[variance_template.array < 0] = 0
+                if np.any(~np.isfinite(variance_template.array)):
+                    if logger:
+                        logger.debug("Setting non-finite-variance pixels to 0 for noise generation.")
+                    variance_template.array[~np.isfinite(variance_template.array)] = 0
+
+                # Randomly vary the injected flux in each pixel,
+                # consistent with the true variance level.
+                rng = galsim.BaseDeviate(noise_seed)
+                variable_noise = galsim.VariableGaussianNoise(rng, variance_template)
+                image_template.addNoise(variable_noise)
+
+                # Set an "estimated" variance level in each pixel,
+                # corresponding to the randomly varied image.
+                variance_template = image_template.copy()
+                variance_template *= variance_scale
+
+            # Add the randomly varied synthetic image to the original
+            # image.
+            common_image += image_template
+            # Add the estimated variance of the injection to the original
+            # variance.
```

Comment:
Do these final in-line comment words really need to be on a separate
line?

I find it somewhat
unsatisfying.

Reply:
Lol.

Reply:
I think black is the culprit here.